internal/driver/glfw/window_desktop.go

```diff
@@ -297,6 +301,13 @@ func (w *window) getMonitorForWindow() *glfw.Monitor {
 }
 
 func (w *window) detectScale() float32 {
+	// check if DPI detection is disabled
+	env := os.Getenv(disableDPIDetectionEnvKey)
+	switch env {
+	case "True", "TRUE", "true", "T", "t", "1":
+		return 1
+	}
+
 	if build.IsWayland { // Wayland controls scale through content scaling
```

Comment:
I think it would be wise to move the env check to after this check. In the case of Wayland, there is no need to check the environment variable at all and the compiler can just deadcode eliminate most of the function.